internal/pkg/agent/application/gateway/fleet/fleet_gateway.go

```diff
@@ -356,17 +359,19 @@ func (f *FleetGateway) execute(ctx context.Context) (*fleetapi.CheckinResponse,
 	resp, took, err := cmd.Execute(ctx, req)
 	if isUnauth(err) {
 		f.unauthCounter++
-
-		if f.shouldUnenroll() {
-			f.log.Warnf("retrieved an invalid api key error '%d' times. Starting to unenroll the elastic agent.", f.unauthCounter)
-			return &fleetapi.CheckinResponse{
-				Actions: []fleetapi.Action{&fleetapi.ActionUnenroll{ActionID: "", ActionType: "UNENROLL", IsDetected: true}},
-			}, took, nil
+		if f.shouldUseLongSched() {
+			f.log.Warnf("retrieved an invalid api key error '%d' times. will use long scheduler", f.unauthCounter)
+			f.scheduler = scheduler.NewPeriodic(defaultGatewaySettings.ErrDuration)
+			return &fleetapi.CheckinResponse{}, took, nil
 		}
 
 		return nil, took, err
 	}
 
+	if _, ok := f.scheduler.(*scheduler.PeriodicJitter); !ok {
```

Comment:
This interface check seems dangerous, if somebody updates a method signature or something this will fail at runtime but not compile time, leaving us stuck on the long scheduler.

Why was this necessary?